tests/test_adaptive_fourier_integral.py

```diff
@@ -0,0 +1,236 @@
+        result = np.zeros_like(x)
+
+        if 2 < x < 3:
+            # Add positive bulge between 2 and 3
+            result +=  - (x-2) * (x-3)
+        elif 6 < x < 7:
+            # Add large negative bulge between 6 and 7
+            result += 3*(x-6) * (x-7)
+        
+        return result
+    
+    func = np.vectorize(func)
+
+    # Absolute error
+    input_error_metric = lambda func_vals, interp_vals: np.abs(func_vals - interp_vals)
+
+
+    output_midpoint_frequencies, output_interpolation_error_by_interval = _find_interval_errors(
+        frequencies=input_frequencies,
+        func_values=func(input_frequencies),
+        func=func,
+        linear_bisection_condition=input_bisection_mode_condition,
+        interpolation_error_metric=input_error_metric,
+        logstep_towards_inf=2 # Not relevant for this test
+    )
+    
+    output_max_error_freq = output_midpoint_frequencies[np.argmax(output_interpolation_error_by_interval)]
+
+    assert output_max_error_freq == pytest.approx(6.5)
+
+    # Since the function is piecewise polynomial with degree <=2, we expect the error to be calculated exactly
+    assert np.sum(output_interpolation_error_by_interval) == pytest.approx(
+        1/6 # small bulge contribution
+        + 3/6 # large bulge contribution
+    )
+
+
+def test_improve_frequency_range():
+    # NOTE: This test does not test what the function *does* (minimize integrated midpoint interpolation error),
+    # but rather what it *is intended to do* (find a frequency grid that one can use as nodes to create a good interpolation)
+    input_frequencies = np.linspace(1.,10.,11)
+    input_finer_frequencies = np.linspace(1.,10.,1000)
+    input_bisection_mode_condition = None
+    input_tolerance = 1e-3
+    
+    # Some random smooth function
+    func = lambda freq: np.sqrt(freq)*np.cos(2*np.pi*freq / 10) + 2*np.exp(-(freq-4)**2 / 0.1)
+
+    # Use absolute error as metric
+    error_metric = lambda func_vals, interp_vals: np.abs(func_vals - interp_vals)
+
+    output_frequencies, output_func_values = improve_frequency_range(
+        initial_frequencies=input_frequencies,
+        func=func, 
+        interpolation_error_metric=error_metric,
+        absolute_integral_tolerance=input_tolerance,
+        logstep_towards_inf=2, # Not relevant for this test
+        bisection_mode_condition=input_bisection_mode_condition
+    )
+
+    output_interpolation_on_fine_grid = complex_pchip(output_frequencies, output_func_values, input_finer_frequencies)
+    expected_interpolation_on_fine_grid = func(input_finer_frequencies)
+
+    # NOTE: Absolute tolerance here because that is used in the add_points... function itself
+    assert simpson(error_metric(output_interpolation_on_fine_grid, expected_interpolation_on_fine_grid), input_finer_frequencies) < input_tolerance
+
+
+@pytest.mark.parametrize(("input_func", "expected_transform"), [
+    ( lambda f: 1 / np.sqrt( 2 * np.pi * f + 1e-50), lambda t: np.sqrt(np.pi / (2 * t)) ),
```

Comment:
here also, one could consider maybe removing `+ 1e-50` ?

Reply:
I disagree, I feel it's important that the test uses `(0, inf)` as the starting range, as I think that will be one of the more commonly used input ranges, so it's nice to test that it doesn't fail. This means that the function must be evaluated at 0, and then to avoid errors this is the simplest way

Reply:
ok, sure, let's keep it as it is.